### PR TITLE
Provide detailed error log upon invalid client configuration

### PR DESCRIPTION
**Pull request by sachinira**
```
ERROR module = ballerina/oauth2 message = "Failed to retrieve access token since the response payload is not a JSON." error = "{ballerina/lang.value}FromJsonStringError"
```
But in a users perspective this error in not informative enough to identify the exact cause for the error. So need to provide more debug information to the user in order to identify the exact error.
Resolves https://github.com/wso2-enterprise/choreo/issues/3635

Then, the error will be look like this. Which enables user to debug it easily
```
time = 2021-05-18T15:06:56.732+05:30 level = ERROR module = ballerina/oauth2 message = "Failed to retrieve access-token since the response payload is not a JSON." error = "{ballerina/lang.value}FromJsonStringError" 
error: Invalid values provided for client configuration parameters.
```

eg: providing invalid refresh URL
https://github.com/wso2-enterprise/choreo/issues/3635

## Automation tests
 - Unit tests 
   > Done
 - Integration tests
   > Done

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2019, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+# Copyright (c) 2021, WSO2 Inc. (http://wso2.com) All Rights Reserved.
```
Comment (Chuhaa) [inline .travis.yml:1]: Is there any specific reason to change the year?

File: Package.md [hunk 2]
```diff
@@ -6,49 +6,58 @@ Connects to Salesforce from Ballerina.
 
 ## Salesforce Data APIs
 
-Salesforce has a vast landscape of APIs since they follow an API-first approach to building features on the Salesforce Platform. This approach gives users flexibility to manipulate their data however they want. The most commonly used Salesforce Data APIs are REST API, SOAP API, Bulk API and Streaming API and together they make up the Salesforce Data API. 
+Salesforce has a vast landscape of APIs since they follow an API-first approach to building features on the Salesforce
```
Comment (Chuhaa) [inline Package.md:9]: Do we need a package.md in root directory? We already have one in /sfdc directory.

File: sfdc/errors.bal [hunk 3]
```diff
@@ -14,15 +13,23 @@
...
+}
+
 // Error constants
 const string JSON_ACCESSING_ERROR_MSG = "Error occurred while accessing the JSON payload of the response.";
 const string XML_ACCESSING_ERROR_MSG = "Error occurred while accessing the XML payload of the response.";
 const string TEXT_ACCESSING_ERROR_MSG = "Error occurred while accessing the Text payload of the response.";
 const string ERR_EXTRACTING_ERROR_MSG = "Error occured while extracting errors from payload.";
 const string HTTP_ERROR_MSG = "Error occurred while getting the HTTP response.";
 const string UNREACHABLE_STATE = "Response type cannot be http payload";
+const string INVLID_CLIENT_CONFIG = "Invalid values provided for client configuration parameters.";
```
Comment (Chuhaa) [inline sfdc/errors.bal:35]: Typo

File: .travis.yml (hunk 1 above)
Comment (sachinira) [reply .travis.yml]: Misunderstood the usage of this year in license header. Reverted it.

File: sfdc/errors.bal (hunk 3 above)
Comment (sachinira) [reply sfdc/errors.bal]: Fixed

**Review by abeykoon — APPROVED** [review]

**Comment by abeykoon** [comment]
We could do formatting fixes in a different PR as it is not in the context of fixing above issue. :)

**Pull request merged**